lib/api/v1/assignment.rb

```diff
@@ -356,8 +356,18 @@ def assignment_json(assignment, user, session, opts = {})
       if submission.is_a?(Array)
         ActiveRecord::Associations::Preloader.new.preload(submission, :quiz_submission) if assignment.quiz?
         hash['submission'] = submission.map { |s| submission_json(s, assignment, user, session, assignment.context, params) }
+        should_show_statistics = opts[:include_score_statistics] && submission.any? { |s| can_show_statistics?(assignment, s, user) }
```

Comment:
It looks like this will lead to an N+1 query problem because Submission#hide_grade_from_student? (invoked on line 406) will make queries. The lines in that method that are currently executing queries are:

1. `return muted_assignment? unless assignment.course.post_policies_enabled?`

Suggestion: the "Post Policies" feature is enabled for everyone in Canvas now, so this line can be safely removed in this commit (we plan to follow up with a commit to remove all checks for this feature flag).

2. `if assignment.post_manually?`

Since the `assignment` association isn't loaded on the submission, this will result in a query to fetch the assignment. However, conveniently, we have the assignment available here, so we just need to manually "load" it onto the submission. On line 405, we should be able to add:

```
submission.assignment = assignment
```

. This way, the `if assignment.post_manually?` line will no longer execute a query to get the assignment (there's some work we'll probably want to do on our side to do make the post_manually? method more performant, but that's outside the scope of your commit).